src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeIndex_Create.cs

```diff
@@ -214,14 +237,69 @@ internal sealed partial class SyntaxTreeIndex
                             containsAwait,
                             containsTupleExpressionOrTupleType),
                     new DeclarationInfo(
-                            declaredSymbolInfos.ToImmutableAndFree()));
+                            declaredSymbolInfos.ToImmutableAndFree()),
+                    new ExtensionMethodInfo(
+                        simpleExtensionMethodInfoBuilder.ToImmutableDictionary(GetKey, GetImmutableValue),
```

Comment:
fyi, these will allocate converting your GetKey and GetImmutableValue to delegates.

Reply:
Ah, right. Fixed